### PR TITLE
fix: Update transaction type for gas estimation if one is not provided for EIP712 transactions, this fixes paymasters for era-test-node.

### PR DESCRIPTION
**Pull request by MexicanAce**
# What :computer: 
* Update transaction type for gas estimation when one is not provided
* Fix the `Debug era_test_node w/ system-contracts` VSCode launch task

# Why :hand:
* These values are required by the Bootloader, else the transaction type will default to Legacy and then fail checks as Legacy transaction types should not have a Paymaster Address
* Useful for local debugging down to the Bootloader and System Contract level

# Evidence :camera:
New e2e tests before:
image

e2e tests after:
image

# Notes :memo:
* Fixes https://github.com/matter-labs/era-test-node/issues/166
* I spent waaaaayyyy too much time trying to figure out unit tests for this change... I'll come back around to try again in a few weeks, but for now an e2e was gonna be necessary anyways, so figure it should suffice for now. The problem is creating a `CallRequest` that can be turned into a valid `TransactionRequest`

### CODE REVIEW

File: e2e-tests/test/erc721gated.test.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    let normalUserWallet = Wallet.createRandom();
+    normalUserWallet = new Wallet(normalUserWallet.privateKey, provider);
+
+    // Act
+    const action = async () => {
+      await executeGreetingTransaction(normalUserWallet, "Hello World");
+    };
+
+    // Assert
+    await expectThrowsAsync(action, "User does not hold the required NFT asset and therefore must for their own gas!");
```
Comment (MexicanAce) [inline e2e-tests/test/erc721gated.test.ts:97]: This is the main check that will ensure we can correctly estimate gas and handle gated checks within Paymasters

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "era_test_node"
-version = "0.1.0-alpha.5"
+version = "0.1.0-alpha.8"
```
Comment (dutterbutter) [inline Cargo.toml:3]: Should change this to `.9` given we will be making a release today or tomorrow?

File: e2e-tests/test/erc721gated.test.ts (hunk 1 above)
Comment (dutterbutter) [reply e2e-tests/test/erc721gated.test.ts:97]: Is it worth testing the `Approval` paymaster flow as well?

**Review by dutterbutter — DISMISSED** [review]
LGTM couple comments.

File: e2e-tests/contracts/ERC20FixedPaymaster.sol [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    function postTransaction(
+        bytes calldata _context,
+        Transaction calldata _transaction,
+        bytes32,
+        bytes32,
+        ExecutionResult _txResult,
+        uint256 _maxRefundedGas
+    ) external payable override onlyBootloader {
+        // Refunds are not supported yet.
```
Comment (dutterbutter) [inline e2e-tests/contracts/ERC20FixedPaymaster.sol:119]: Nit: can you remove this comment, need to remove from paymaster-examples as well. Causes too much confusion.

File: src/node.rs [hunk 4]
```diff
@@ -49,18 +49,19 @@ use zksync_core::api_server::web3::backend_jsonrpc::{
...
     fee::Fee,
     get_code_key, get_nonce_key,
     l2::L2Tx,
+    l2::TransactionType,
     transaction_request::TransactionRequest,
     utils::{
         decompose_full_nonce, nonces_to_full_nonce, storage_key_for_eth_balance,
         storage_key_for_standard_token_balance,
     },
+    vm_trace::Call,
```
Comment (dutterbutter) [inline src/node.rs:64]: Where is this used?

Comment (MexicanAce) [reply src/node.rs:64]: Various places. This `use` statement was just combined. Wasn't sure why it was separated in the first place

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**